Breeze.TumbleBit.Client/TumbleBitManager.cs

```diff
@@ -353,29 +351,46 @@ public enum TumbleState
                     this.logger.LogDebug("Not enough masternode registrations downloaded yet: " + registrations.Count);
                     return Result.Fail<ClassicTumblerParameters>("Not enough masternode registrations downloaded yet");
                 }
-                
+
                 registrations.Shuffle();
 
-                // Since the list is shuffled, we can simply iterate through it and try each server until one is valid & reachable
+                // Since the list is shuffled, we can simply iterate through it and try each server until one is valid & reachable.
                 foreach (RegistrationRecord record in registrations)
                 {
                     this.TumblerAddress = "ctb://" + record.Record.OnionAddress + ".onion?h=" + record.Record.ConfigurationHash;
 
-                    var attemptConnection = await TryUseServer();
+                    try
+                    {
+                        var attemptConnection = await TryUseServer();
 
-                    if (!attemptConnection.Failure)

```

Comment:
Result object should have `Success` boolean, so instead of double negative, we should change this to `if (!attemptConnection.Success)`